lib/imgix/rails/tag.rb

```diff
@@ -31,127 +31,43 @@ def srcset(url_params: @url_params, widths: @widths)
-  IPHONE_6_PLUS_LANDSCAPE = { css_width: 736, dpr: 3 }
-  MOTO_NEXUS_6 = { css_width: 412, dpr: 3.5 }
-  MOTO_NEXUS_6_LANDSCAPE = { css_width: 690, dpr: 3.5 }
-  LUMIA_1520 = { css_width: 432, dpr: 2.5 }
-  LUMIA_1520_LANDSCAPE = { css_width: 768, dpr: 2.5 }
-  GALAXY_NOTE_3 = { css_width: 360, dpr: 3 }
-  GALAXY_NOTE_3_LANDSCAPE = { css_width: 640, dpr: 3 }
-  GALAXY_NOTE_4 = { css_width: 360, dpr: 4 }
-  GALAXY_NOTE_4_LANDSCAPE = { css_width: 640, dpr: 4 }
-
-  # Tablets
-  IPAD = { css_width: 768, dpr: 1 };
-  IPAD_LANDSCAPE = { css_width: 1024, dpr: 1 };
-  IPAD_3 = { css_width: 768, dpr: 2 };
-  IPAD_3_LANDSCAPE = { css_width: 1024, dpr: 2 };
-  IPAD_PRO = { css_width: 1024, dpr: 2 };
-  IPAD_PRO_LANDSCAPE = { css_width: 1366, dpr: 2 };
-
-  BOOTSTRAP_SM = { css_width: 576, dpr: 1 }
-  BOOTSTRAP_MD = { css_width: 720, dpr: 1 }
-  BOOTSTRAP_LG = { css_width: 940, dpr: 1 }
-  BOOTSTRAP_XL = { css_width: 1140, dpr: 1 }
-
-  def devices
-    phones + phablets + tablets + bootstrap_breaks
-  end
-
-  def bootstrap_breaks
-    breaks = [
-      BOOTSTRAP_SM,
-      BOOTSTRAP_MD,
-      BOOTSTRAP_LG,
-      BOOTSTRAP_XL
-    ]
+  INCREMENT_PRECENTAGE = 8
 
-    breaks + breaks.map { |b| b[:dpr] = 2; b }
-  end
+  @@standard_widths = nil
 
-  def phones
-    [
-      IPHONE,
-      IPHONE_4,
-      IPHONE_6,
-      LG_G3
-    ]
-  end
+  def standard_widths
+    return @@standard_widths if @@standard_widths
 
-  def phablets
-    [
-      IPHONE_6_PLUS,
-      IPHONE_6_PLUS_LANDSCAPE,
-      MOTO_NEXUS_6,
-      MOTO_NEXUS_6_LANDSCAPE,
-      LUMIA_1520,
-      LUMIA_1520_LANDSCAPE,
-      GALAXY_NOTE_3,
-      GALAXY_NOTE_3_LANDSCAPE,
-      GALAXY_NOTE_4,
-      GALAXY_NOTE_4_LANDSCAPE
-    ]
-  end
+    prev = MINIMUM_SCREEN_WIDTH
+    @@standard_widths = []
+    while prev < MAXIMUM_SCREEN_WIDTH do
+      @@standard_widths.append(2 * (prev/2).round)   # Ensure widths are even
+      prev = prev * (1 + (INCREMENT_PRECENTAGE/100.0)*2.0)
+    end
+    @@standard_widths.push(MAXIMUM_SCREEN_WIDTH)
+    @@standard_widths.freeze
 
-  def tablets
-    [
-      IPAD,
-      IPAD_LANDSCAPE,
-      IPAD_3,
-      IPAD_3_LANDSCAPE,
-      IPAD_PRO,
-      IPAD_PRO_LANDSCAPE
-    ]
+    @@standard_widths
   end
 
   # Return the widths to generate given the input `sizes`
   # attribute.
   #
   # @return {Array} An array of {Fixnum} instances representing the unique `srcset` URLs to generate.
   def target_widths
-    min_screen_width_required = @tag_options[:min_width] || SCREEN_STEP
-    max_screen_width_required = @tag_options[:max_width] || MAXIMUM_SCREEN_WIDTH
-
-    widths = (device_widths + screen_widths).select do |w|
-      w <= max_screen_width_required && w >= min_screen_width_required
-    end.compact.uniq.sort
-
-    # Add exact widths for 1x, 2x, and 3x devices
-    if @url_params[:w]
-      widths.push(@url_params[:w], @url_params[:w] * 2, @url_params[:w] * 3)
+    min_width = @tag_options[:min_width]
+    max_width = @tag_options[:max_width]
```

Comment:
Would it be too silly to add a check here to see if `min_width` was larger than `max_width`, and warn the user if it was? What about checking if `min_width` and `max_width` are inside `(0, 5120]`?

Also, what if a user picks a `max_width` higher than 5120? Do we want to generate widths up to that value? @jayeb can probably weigh in here. What about for a `min_width` less than 100px?

Reply:
I think we should allow `min_width` down to 0 and `max_width` up to 8192 (the maximum width imgix can render).

Reply:
Crazy idea... should we make 8192 the default max_width? @jayeb

Reply:
> What about checking if `min_width` and `max_width` are inside (0, 5120]

IMO, the library shouldn't enforce a restriction on the upper-limit. Let's trust the developers know what they are doing with a `max_width` greater than our limit.